template.rb

```diff
@@ -0,0 +1,123 @@
+require 'bundler'
+
+def install
+  add_template_repository_to_source_path
+  install_gems
+  install_solidus
+  copy_solidus_starter_frontend_files
+  update_asset_files
+  require_solidus_starter_frontend_config
+  install_rspec
+end
+
+# Copied from: https://github.com/mattbrictson/rails-template
+# Add this template directory to source_paths so that Thor actions like
+# copy_file and template resolve against our source files. If this file was
+# invoked remotely via HTTP, that means the files are not present locally.
+# In that case, use `git clone` to download them to a local temporary dir.
+def add_template_repository_to_source_path
+  if __FILE__ =~ %r{\Ahttps?://}
+    require "tmpdir"
+
+    tempdir = Dir.mktmpdir("solidus_starter_frontend-")
+    repo_dir = tempdir
+
+    at_exit { FileUtils.remove_entry(tempdir) }
+
+    git clone: [
+      "--quiet",
+      "https://github.com/nebulab/solidus_starter_frontend.git",
+      tempdir
+    ].map(&:shellescape).join(" ")
+
+    if (branch = __FILE__[%r{solidus_starter_frontend/(.+)/template.rb}, 1])
+      Dir.chdir(tempdir) { git checkout: branch }
+    end
+  else
+    repo_dir = File.dirname(__FILE__)
+  end
+
+  templates_dir = Pathname.new(repo_dir).join('templates')
+
+  source_paths.unshift(templates_dir)
+end
+
+def install_gems
+  add_solidus_gems
+  add_solidus_starter_frontend_dependencies
+  add_spec_gems
+
+  Bundler.with_original_env do
+    run 'bundle install'
+  end
+end
+
+def add_solidus_gems
+  solidus_repo = ENV['SOLIDUS_REPO']
+  solidus_branch = ENV['SOLIDUS_BRANCH']
+
+  if solidus_repo && solidus_branch
```

Comment:
@aldesantis FYI, I added this `solidus_repo` check after I made the walkthrough. Previously, the code only checked `solidus_branch` and set `solidus_repo` to the default only if `solidus_branch` is provided. You can see the previous code in the walkthrough.